# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/RehoPrince/Flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

I chose the `Ranking Signal Analysis` lane. I want to examine which observable content and performance signals are associated with recent impression declines, then use those associations to support content-prioritization decisions. This fits the starter data because each row contains a content item, its trailing 90-day metrics, and a recent-versus-previous 30-day trend summary.

In [1]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

- **Decision:** Which content pages should an SEO strategist or content operations manager review first?
- **Action:** Use measured signals to prioritize pages for investigation and possible refresh, rather than treating the score as an automatic rewrite or deletion decision.
- **Cost of a wrong call:** A false positive wastes optimization effort on a page that does not need attention. A false negative delays action on a page whose recent performance is weakening.

In [2]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [3]:
import pandas as pd

DF_PATH = "../../data/raw/content_refresh_anonymized.csv"
df = pd.read_csv(DF_PATH)

df.columns

Index(['content_id', 'client_id', 'search_volume', 'competition',
       'competition_level', 'cpc', 'content_type', 'main_intent', 'word_count',
       'char_count', 'provider_used', 'model_used', 'impressions_90d',
       'clicks_90d', 'pageviews_90d', 'sessions_90d', 'users_90d',
       'engaged_sessions_90d', 'ai_sessions_90d', 'scroll_events_90d',
       'days_with_impressions', 'days_with_sessions', 'impressions_last_30d',
       'clicks_last_30d', 'sessions_last_30d', 'impressions_prev_30d',
       'clicks_prev_30d', 'sessions_prev_30d', 'content_age_days', 'age_tier',
       'age_tier_order', 'days_since_last_update', 'freshness_tier',
       'word_count_tier', 'char_count_tier', 'ctr', 'avg_position',
       'engagement_rate', 'scroll_rate', 'ai_traffic_pct', 'impression_tier',
       'position_tier', 'trend_direction', 'trend_pct'],
      dtype='str')

In [4]:
down_trend_count = (df["trend_direction"] == "down").sum()

print("Rows: ", df.shape[0])
print("Clients: ", df["client_id"].nunique())
print("Recent-impression down items: ", down_trend_count)
print("Recent-impression down share: ", round(down_trend_count / len(df), 3))

Rows:  30000
Clients:  32
Recent-impression down items:  16262
Recent-impression down share:  0.542


Across 30,000 content items from 32 pseudonymized clients, 16,262 items (54.2%) show a down trend.

This means their impressions were lower in the most recent 30-day window than in the previous 30-day window.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

I can claim that the analysis measures associations between available content/performance signals and the observed recent-impression trend in this starter snapshot. It may support  prioritization for pages that deserve review.

I cannot claim that any signal causes traffic changes, that the analysis will drive Google traffic, or that it guarantees customer conversion.

In [5]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.